internal/planner/se_scan.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package planner
+
+import (
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/client/request"
+	"github.com/sourcenetwork/defradb/internal/keys"
+	"github.com/sourcenetwork/defradb/internal/planner/mapper"
+	"github.com/sourcenetwork/defradb/internal/se"
+)
+
+// seScanNode implements a plan node for searchable encryption queries.
+// It queries remote nodes for document IDs matching the search criteria
+// and returns only the document IDs.
+type seScanNode struct {
+	documentIterator
+	docMapper
+
+	p                *Planner
+	collection       client.Collection
+	collectionID     string
+	filter           *mapper.Filter
+	encryptedIndexes []client.EncryptedIndexDescription
+
+	fieldSearchTags map[string][]byte
+	remoteDocIDs    []string
+	hasReturned     bool
+}
+
+var _ planNode = (*seScanNode)(nil)
+
+func (n *seScanNode) Kind() string { return "seScanNode" }
+
+func (n *seScanNode) Init() error {
+	return nil
+}
+
+func (n *seScanNode) Start() error {
+	if err := n.generateSearchTags(); err != nil {
+		return err
+	}
+
+	n.remoteDocIDs = nil
+	n.hasReturned = false
+
+	return nil
+}
+
+func (n *seScanNode) generateSearchTags() error {
+	n.fieldSearchTags = make(map[string][]byte)
+
+	for fieldName, condition := range n.filter.ExternalConditions {
+		var encIdx *client.EncryptedIndexDescription
+		for _, idx := range n.encryptedIndexes {
+			if idx.FieldName == fieldName {
+				encIdx = &idx
+				break
+			}
+		}
+
+		if encIdx == nil {
+			continue
+		}
+
+		condMap, ok := condition.(map[string]any)
+		if !ok {
+			return NewErrInvalidEncryptedFieldCondition(fieldName)
```

Comment:
question: How is this possible?

todo: If it is not, please remove it.  If it is, please add an integration test that covers it.